File: Examples/DotNET/VisualBasic/Document/AppendPdfFiles.vb
Imports Com.Aspose.PDF.Api
Imports Com.Aspose.PDF.Model
Imports Com.Aspose.Storage.Api

Namespace Document
    Class AppendPdfFiles
        Public Shared Sub Run()
            ' ExStart:1
            Dim pdfApi As New PdfApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH)
            Dim storageApi As New StorageApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH)

            Dim fileName As [String] = "Sample.pdf"
            Dim appendFile As [String] = Nothing
            Dim startPage As Integer = 2
            Dim endPage As Integer = 3
            Dim storage As [String] = ""
            Dim folder As [String] = ""

            Dim appendFileName As [String] = "sample-input.pdf"
            Dim body As New AppendDocument()
            body.Document = appendFileName
            body.StartPage = 2
            body.EndPage = 3

            Try
                ' Upload source file to aspose cloud storage
                storageApi.PutCreate(fileName, "", "", System.IO.File.ReadAllBytes(Common.GetDataDir() + fileName))
                storageApi.PutCreate(appendFileName, "", "", System.IO.File.ReadAllBytes(Common.GetDataDir() + appendFileName))

                ' Invoke Aspose.PDF Cloud SDK API to append pdf file
                Dim apiResponse As DocumentResponse = pdfApi.PostAppendDocument(fileName, appendFile, startPage, endPage, storage, folder, _
                    body)

                If apiResponse IsNot Nothing AndAlso apiResponse.Status.Equals("OK") Then
                    Console.WriteLine("Append PDF Files, Done!")
                    Console.ReadKey()
                End If
            Catch ex As Exception
                System.Diagnostics.Debug.WriteLine("error:" + ex.Message + vbLf + ex.StackTrace)
            End Try
            ' ExEnd:1
        End Sub
    End Class
End Namespace

File: Examples/DotNET/CSharp/Document/AppendPdfFiles.cs
using System;
using Com.Aspose.PDF.Api;
using Com.Aspose.PDF.Model;
using Com.Aspose.Storage.Api;

namespace Document
{
    class AppendPdfFiles
    {
        public static void Run()
        {
            // ExStart:1
            PdfApi pdfApi = new PdfApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);
            StorageApi storageApi = new StorageApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);

            String fileName = "Sample.pdf";
            String appendFile = null;
            int startPage = 2;
            int endPage = 3;
            String storage = "";
            String folder = "";

            String appendFileName = "sample-input.pdf";
            AppendDocument body = new AppendDocument();
            body.Document = appendFileName;
            body.StartPage = 2;
            body.EndPage = 3;

            try
            {
                // Upload source file to aspose cloud storage
                storageApi.PutCreate(fileName, "", "", System.IO.File.ReadAllBytes(Common.GetDataDir() + fileName));
                storageApi.PutCreate(appendFileName, "", "", System.IO.File.ReadAllBytes(Common.GetDataDir() + appendFileName));

                // Invoke Aspose.PDF Cloud SDK API to append pdf file
                DocumentResponse apiResponse = pdfApi.PostAppendDocument(fileName, appendFile, startPage, endPage, storage, folder, body);

                if (apiResponse != null && apiResponse.Status.Equals("OK"))
                {
                    Console.WriteLine("Append PDF Files, Done!");
                    Console.ReadKey();
                }
            }
            catch (Exception ex)
            {
                System.Diagnostics.Debug.WriteLine("error:" + ex.Message + "\n" + ex.StackTrace);
            }
            // ExEnd:1
        }
    }
}
